test/directives/prefetch-llvm_13.f90

```diff
@@ -0,0 +1,25 @@
+! REQUIRES: llvm-13
+
+! RUN: %flang -O1 -S -emit-llvm %s -o - | FileCheck %s --check-prefix=CHECK-PREFETCH
+! RUN: %flang -O1 -S -emit-llvm -Hy,59,0x4 %s -o - | FileCheck %s --check-prefix=CHECK-NOPREFETCH
```

Comment:
I still don't understand why we cannot have just one test, by making the `RUN` commands use `-O0`. I have verified that, at `-O0`, both LLVM 12 and LLVM 13 will produce the exact same IR, including both `llvm.prefetch` intrinsic calls as expected. The only change you need is matching a signed extension and multiplication of the 32-bit loop induction variable, instead of a left-shift of the 64-bit version of the same variable. For example:
```
! CHECK-PREFETCH: [[i:%[0-9]+]] = load i32
! CHECK-PREFETCH: [[TMP1:%[0-9]+]] = sext i32 [[i]] to i64
! CHECK-PREFETCH: [[TMP2:%[0-9]+]] = mul nsw i64 [[TMP1]], 4
! CHECK-PREFETCH: [[a2elem:%[0-9]+]] = getelementptr i8, i8* [[a2base]], i64 [[TMP2]]
! CHECK-PREFETCH: call void @llvm.prefetch{{.*}}(i8* [[a2elem]], i32 0, i32 3, i32 1)
```

Reply:
let's try...